base/ca/src/main/java/com/netscape/cms/profile/common/CAEnrollProfile.java

```diff
@@ -531,4 +532,113 @@ public void execute(Request request) throws EProfileException, ERejectException
         }
     }
 
+    private Map<String, byte[]> convertP12Password (Request request) throws EProfileException {
+        String method = "CAEnrollProfile: convertP12Password: ";
+        Map<String, byte[]> returnPass = null;
+
+        String p12passwd = request.getExtDataInString("serverSideKeygenP12Passwd");
+
+        org.mozilla.jss.crypto.X509Certificate transCert = null;
+
+        CAEngine engine = CAEngine.getInstance();
+        CertificateAuthority ca = engine.getCA();
+        CAConfig caConfig = ca.getConfigStore();
+        ConnectorsConfig connectorsConfig = caConfig.getConnectorsConfig();
+        ConnectorConfig kraConnectorConfig = connectorsConfig.getConnectorConfig("KRA");
+
+        try {
+            CryptoManager cm = CryptoManager.getInstance();
+            String transportNickname = kraConnectorConfig.getString("transportCertNickname", "KRA Transport Certificate");
+            transCert = cm.findCertByNickname(transportNickname);
+        } catch (Exception e) {
+            logger.debug(method + "'KRA transport certificate' not found in nssdb; need to be manually setup for Server-Side keygen enrollment");
+            throw new EProfileException(CMS.getUserMessage("CMS_MISSING_KRA_TRANSPORT_CERT_IN_CA_NSSDB"));
+
+            /* future; cert nickname can't be controlled yet at import in jss
+            logger.debug(method + "KRA transport certificate not found in nssdb; getting from CS.cfg");
+            transportCertStr = connectorsConfig.getString("KRA.transportCert", "");
+            logger.debug(method + "transportCert found in CS.cfg: " + transportCertStr);
+
+            byte[] transportCertB = Utils.base64decode(transportCertStr);
+            logger.debug(method + "transportCertB.length=" + transportCertB.length);
+            // hmmm, can't yet control the nickname
+            transCert = cm.importCACertPackage(transportCertB);
+            logger.debug(method + "KRA transport certificate imported");
+            */
```

Comment:
I don't think we should support storing the transport cert in `CS.cfg` anymore. The transport cert should already be stored in NSS database when the server is started and we should not assume we can modify the NSS database when the server is running.